### PR TITLE
[10% review] Bug 1434021 - use r.taskcluster.net for docs redirects

### PR DESCRIPTION
**Pull request by djmitche**
I'm really not sure I like this approach.  The idea is that in the docker image, all links are absolute and point to `https://r.taskcluster.net`.  Then during startup of the container (or in nginx) those would be rewritten to the rootUrl.  Completing this would entail writing a `taskcluster-redirector` service that could run at `r.taskcluster.net` in heroku and redirect `https://r.taskcluster.net/docs/$path` to `https://docs.taskcluster.net/$path`.

Here's the things I want to continue to work:

* Existing, widely-distributed links to URLs on https://docs.taskcluster.net
* Clicking links in a README on Github works -- for example, `https://r.taskcluster.net/docs/manual/design/apis/hawk/scopes`.
* Clicking links in a README in the reference section of a deployment works, and stays within the same rootUrl (so a link on https://tc.example.com/docs/reference/foo/bar would point to https://tc.example.com/docs/bing/baz).
* Clicking links in a README in the reference section of the old site (docs.taskcluster.net) still works (redirecting is OK)

This does sort of commit us to running r.taskcluster.net forever.

If we didn't copy READMEs, but instead read `docs/index.md` for example, this would get simpler because links in Github wouldn't have to work.

Basically I feel like this is a dirty hack and a lot of work, but I don't have better ideas...

### CODE REVIEW

File: lib/absurls.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ * anything missing that prefix.
+ *
+ * These URLs will be rewritten on deployment to change `r.taskcluster.net`
+ * to the hostname of the deployment's rootUrl. URLs are made absolute to
+ * make that rewrite easier (done with a regex instead of an HTML parser).
+ *
+ * Eventually, all documentation should use either absolute `r.taskcluster.net`
+ * URLs or, where possible, relative URLs.
+ */
+exports.rewriteAbsoluteUrls = function() {
```
[the remark this answers is not in the document]
Comment (eliperelman) [reply lib/absurls.js]: Haha I have, just didn't know if there was a better reason. 😆

**Comment by imbstack** [comment]
Let's just not copy READMEs. It was always kinda a hack and it ends up being a weird interface anyway. We could do another round of updates to all of our services where we basically copy-paste READMEs in to `docs/index.md`, remove a few bits that no longer make sense and then move on with our lives?

**Comment by imbstack** [comment]
Which is to say I haven't reviewed the rest of this yet, but if not copying READMEs makes your life easier, let's do it.

**Comment by djmitche** [comment]
That sounds like a good place to start.  I'll hack s'more on this.
